From e17d174510990c6c127887efc9696b41103e2c9f Mon Sep 17 00:00:00 2001
From: "Chen, Haochuan" <haochuan.z.chen@intel.com>
Date: Sun, 31 Dec 2023 14:02:57 +0800
Subject: [PATCH] Enable multi-zone audio for VM3

add audio_policy_configuration.xml and car_audio_configuration.xml

Tracked-On: OAM-114563
Change-Id: I0fcc90fb216037993d2657e73617c6efc9222e2d
Signed-off-by: Chen, Haochuan <haochuan.z.chen@intel.com>
---
 .../audio/audio_policy_configuration.xml      |   0
 ..._policy_configuration_attached_devices.xml |  44 ++++
 .../audio_policy_configuration_devices.xml    | 224 ++++++++++++++++++
 .../audio_policy_configuration_mixports.xml   | 117 +++++++++
 .../audio_policy_configuration_routes.xml     |  48 ++++
 .../audio/audio_policy_configuration_vm3.xml  |  81 +++++++
 .../audio/car_audio_configuration_vm3.xml     | 110 +++++++++
 groups/multi-passenger/true/product.mk        |  14 ++
 8 files changed, 638 insertions(+)
 mode change 100755 => 100644 groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration.xml
 create mode 100644 groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_attached_devices.xml
 create mode 100644 groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_devices.xml
 create mode 100644 groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_mixports.xml
 create mode 100644 groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_routes.xml
 create mode 100644 groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_vm3.xml
 create mode 100644 groups/multi-passenger/true/overlay-multi_passenger/audio/car_audio_configuration_vm3.xml

diff --git a/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration.xml b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration.xml
old mode 100755
new mode 100644
diff --git a/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_attached_devices.xml b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_attached_devices.xml
new file mode 100644
index 0000000..04aab60
--- /dev/null
+++ b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_attached_devices.xml
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright (C) 2018 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<attachedDevices>
+    <!-- One bus per context -->
+    <item>bus0_media_out</item>
+    <item>bus1_navigation_out</item>
+    <item>bus2_voice_command_out</item>
+    <item>bus3_call_ring_out</item>
+    <item>bus4_call_out</item>
+    <item>bus5_alarm_out</item>
+    <item>bus6_notification_out</item>
+    <item>bus7_system_sound_out</item>
+
+    <!-- TODO update for input-->
+    <item>bus_1000_input_zone_0</item>
+    <item>Built-In Mic</item>
+    <item>Built-In Back Mic</item>
+    <item>Echo-Reference Mic</item>
+
+    <item>bus10_media_out</item>
+    <item>bus11_navigation_out</item>
+    <item>bus12_voice_command_out</item>
+    <item>bus13_call_ring_out</item>
+    <item>bus14_call_out</item>
+    <item>bus15_alarm_out</item>
+    <item>bus16_notification_out</item>
+    <item>bus17_system_sound_out</item>
+
+    <!-- TODO update for input-->
+    <item>bus_2000_input_zone_1</item>
+</attachedDevices>
diff --git a/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_devices.xml b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_devices.xml
new file mode 100644
index 0000000..2a4cd04
--- /dev/null
+++ b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_devices.xml
@@ -0,0 +1,224 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright (C) 2018 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<devicePorts>
+    <!-- zone 1 -->
+    <devicePort tagName="bus0_media_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus0_media_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+            </gains>
+    </devicePort>
+    <devicePort tagName="bus1_navigation_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus1_navigation_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus2_voice_command_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus2_voice_command_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus3_call_ring_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus3_call_ring_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus4_call_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus4_call_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                             samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus5_alarm_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus5_alarm_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                             samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus6_notification_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus6_notification_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus7_system_sound_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus7_system_sound_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+
+    <!-- TODO update for input-->
+    <devicePort tagName="bus_1000_input_zone_0" type="AUDIO_DEVICE_IN_BUS"
+                role="source" address="bus_1000_input_zone_0">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_IN_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="Built-In Mic" type="AUDIO_DEVICE_IN_BUILTIN_MIC" role="source"
+                address="Built-In Mic" >
+        <profile name="" format="AUDIO_FORMAT_PCM_16_BIT"
+                 samplingRates="8000,11025,12000,16000,22050,24000,32000,44100,48000"
+                 channelMasks="AUDIO_CHANNEL_IN_MONO,AUDIO_CHANNEL_IN_STEREO,AUDIO_CHANNEL_IN_FRONT_BACK"/>
+    </devicePort>
+    <devicePort tagName="Built-In Back Mic" type="AUDIO_DEVICE_IN_BACK_MIC"
+                role="source" address="Built-In Back Mic">
+        <profile name="" format="AUDIO_FORMAT_PCM_16_BIT"
+                 samplingRates="8000,11025,12000,16000,22050,24000,32000,44100,48000"
+                 channelMasks="AUDIO_CHANNEL_IN_MONO,AUDIO_CHANNEL_IN_STEREO,AUDIO_CHANNEL_IN_FRONT_BACK"/>
+    </devicePort>
+    <devicePort tagName="Echo-Reference Mic" type="AUDIO_DEVICE_IN_ECHO_REFERENCE" role="source"
+                address="Echo-Reference Mic">
+        <profile name="" format="AUDIO_FORMAT_PCM_16_BIT"
+                 samplingRates="8000,11025,12000,16000,22050,24000,32000,44100,48000"
+                 channelMasks="AUDIO_CHANNEL_IN_MONO,AUDIO_CHANNEL_IN_STEREO,AUDIO_CHANNEL_IN_FRONT_BACK"/>
+    </devicePort>
+
+    <!-- zone 2 -->
+    <devicePort tagName="bus10_media_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus10_media_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+            </gains>
+    </devicePort>
+    <devicePort tagName="bus11_navigation_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus11_navigation_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus12_voice_command_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus12_voice_command_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus13_call_ring_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus13_call_ring_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus14_call_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus14_call_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                             samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus15_alarm_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus15_alarm_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                             samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus16_notification_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus16_notification_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+    <devicePort tagName="bus17_system_sound_out" role="sink" type="AUDIO_DEVICE_OUT_BUS"
+                address="bus17_system_sound_out">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+
+    <!-- TODO update for input-->
+    <devicePort tagName="bus_2000_input_zone_1" type="AUDIO_DEVICE_IN_BUS"
+                role="source" address="bus_2000_input_zone_1">
+        <profile name="" format="AUDIO_FORMAT_PCM_16_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_IN_STEREO"/>
+        <gains>
+            <gain name="" mode="AUDIO_GAIN_MODE_JOINT"
+                  minValueMB="-3200" maxValueMB="600"
+                  defaultValueMB="0" stepValueMB="100"/>
+        </gains>
+    </devicePort>
+</devicePorts>
diff --git a/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_mixports.xml b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_mixports.xml
new file mode 100644
index 0000000..b1f3e7c
--- /dev/null
+++ b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_mixports.xml
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright (C) 2018 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<mixPorts>
+    <mixPort name="mixport_bus0_media_out" role="source"
+             flags="AUDIO_OUTPUT_FLAG_PRIMARY">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        </mixPort>
+    <mixPort name="mixport_bus1_navigation_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus2_voice_command_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus3_call_ring_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus4_call_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus5_alarm_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus6_notification_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus7_system_sound_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <!-- TODO update for input-->
+    <mixPort name="primary input" role="sink">
+        <profile name="" format="AUDIO_FORMAT_PCM_16_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_IN_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus_1000_input_zone_0" role="sink">
+        <profile name="" format="AUDIO_FORMAT_PCM_16_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_IN_STEREO"/>
+    </mixPort>
+
+    <mixPort name="mixport_bus10_media_out" role="source"
+             flags="AUDIO_OUTPUT_FLAG_PRIMARY">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+        </mixPort>
+    <mixPort name="mixport_bus11_navigation_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus12_voice_command_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus13_call_ring_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus14_call_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus15_alarm_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus16_notification_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <mixPort name="mixport_bus17_system_sound_out" role="source">
+        <profile name="" format="AUDIO_FORMAT_PCM_32_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+    </mixPort>
+    <!-- TODO update for input-->
+    <mixPort name="mixport_bus_2000_input_zone_1" role="sink">
+        <profile name="" format="AUDIO_FORMAT_PCM_16_BIT"
+                 samplingRates="48000"
+                 channelMasks="AUDIO_CHANNEL_IN_STEREO"/>
+    </mixPort>
+</mixPorts>
diff --git a/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_routes.xml b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_routes.xml
new file mode 100644
index 0000000..52bdbf7
--- /dev/null
+++ b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_routes.xml
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright (C) 2018 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<routes>
+    <route type="mix" sink="bus10_media_out" sources="mixport_bus10_media_out"/>
+    <route type="mix" sink="bus11_navigation_out" sources="mixport_bus11_navigation_out"/>
+    <route type="mix" sink="bus12_voice_command_out"
+           sources="mixport_bus12_voice_command_out"/>
+    <route type="mix" sink="bus13_call_ring_out" sources="mixport_bus13_call_ring_out"/>
+    <route type="mix" sink="bus14_call_out" sources="mixport_bus14_call_out"/>
+    <route type="mix" sink="bus15_alarm_out" sources="mixport_bus15_alarm_out"/>
+    <route type="mix" sink="bus16_notification_out"
+           sources="mixport_bus16_notification_out"/>
+    <route type="mix" sink="bus17_system_sound_out"
+           sources="mixport_bus17_system_sound_out"/>
+
+    <!-- TODO update for input-->
+    <route type="mix" sink="mixport_bus_2000_input_zone_1" sources="bus_2000_input_zone_1"/>
+
+    <route type="mix" sink="bus0_media_out" sources="mixport_bus0_media_out"/>
+    <route type="mix" sink="bus1_navigation_out" sources="mixport_bus1_navigation_out"/>
+    <route type="mix" sink="bus2_voice_command_out"
+           sources="mixport_bus2_voice_command_out"/>
+    <route type="mix" sink="bus3_call_ring_out" sources="mixport_bus3_call_ring_out"/>
+    <route type="mix" sink="bus4_call_out" sources="mixport_bus4_call_out"/>
+    <route type="mix" sink="bus5_alarm_out" sources="mixport_bus5_alarm_out"/>
+    <route type="mix" sink="bus6_notification_out"
+           sources="mixport_bus6_notification_out"/>
+    <route type="mix" sink="bus7_system_sound_out"
+           sources="mixport_bus7_system_sound_out"/>
+
+    <!-- TODO update for input-->
+    <route type="mix" sink="mixport_bus_1000_input_zone_0" sources="bus_1000_input_zone_0"/>
+    <route type="mix" sink="primary input"
+           sources="Built-In Mic,Built-In Back Mic,Echo-Reference Mic"/>
+</routes>
diff --git a/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_vm3.xml b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_vm3.xml
new file mode 100644
index 0000000..a50ab34
--- /dev/null
+++ b/groups/multi-passenger/true/overlay-multi_passenger/audio/audio_policy_configuration_vm3.xml
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!-- Copyright (C) 2018 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+
+<audioPolicyConfiguration version="1.0" xmlns:xi="http://www.w3.org/2001/XInclude">
+    <!-- version section contains a “version” tag in the form “major.minor” e.g version=”1.0” -->
+
+    <!-- Global configuration Decalaration -->
+    <globalConfiguration speaker_drc_enabled="true"/>
+
+    <!-- Modules section:
+        There is one section per audio HW module present on the platform.
+        Each module section will contains two mandatory tags for audio HAL “halVersion” and “name”.
+        The module names are the same as in current .conf file:
+                “primary”, “A2DP”, “remote_submix”, “USB”
+        Each module will contain the following sections:
+        “devicePorts”: a list of device descriptors for all input and output devices accessible via
+        this module.
+        This contains both permanently attached devices and removable devices.
+            "gain": constraints applied to the millibel values:
+                - maxValueMB >= minValueMB
+                - defaultValueMB >= minValueMB && defaultValueMB <= maxValueMB
+                - (maxValueMB - minValueMB) % stepValueMB == 0
+                - (defaultValueMB - minValueMB) % stepValueMB == 0
+        “mixPorts”: listing all output and input streams exposed by the audio HAL
+        “routes”: list of possible connections between input and output devices or between stream
+        and devices.
+            "route": is defined by an attribute:
+                -"type": <mux|mix> means all sources are mutual exclusive (mux) or can be mixed (mix)
+                -"sink": the sink involved in this route
+                -"sources": all the sources than can be connected to the sink via vis route
+        “attachedDevices”: permanently attached devices.
+        The attachedDevices section is a list of devices names. The names correspond to device names
+        defined in <devicePorts> section.
+        “defaultOutputDevice”: device to be used by default when no policy rule applies
+    -->
+    <modules>
+        <!-- Primary Audio HAL -->
+        <module name="primary" halVersion="3.0">
+            <!-- TODO to be considered for restructuring in Base AAOS -->
+            <defaultOutputDevice>bus0_media_out</defaultOutputDevice>
+            <xi:include href="audio_policy_configuration_attached_devices.xml"/>
+            <xi:include href="audio_policy_configuration_mixports.xml"/>
+            <xi:include href="audio_policy_configuration_devices.xml"/>
+	    <xi:include href="audio_policy_configuration_routes.xml"/>
+        </module>
+
+        <!-- A2dp Audio HAL -->
+        <xi:include href="a2dp_audio_policy_configuration.xml"/>
+
+        <!-- Usb Audio HAL -->
+        <!--xi:include href="usb_audio_policy_configuration.xml"/-->
+
+        <!-- Remote Submix Audio HAL -->
+        <xi:include href="r_submix_audio_policy_configuration.xml"/>
+
+    </modules>
+    <!-- End of Modules section -->
+
+    <!-- Volume section -->
+
+    <xi:include href="audio_policy_volumes.xml"/>
+    <xi:include href="default_volume_tables.xml"/>
+
+    <!-- End of Volume section -->
+    <!-- End of Modules section -->
+
+</audioPolicyConfiguration>
+
diff --git a/groups/multi-passenger/true/overlay-multi_passenger/audio/car_audio_configuration_vm3.xml b/groups/multi-passenger/true/overlay-multi_passenger/audio/car_audio_configuration_vm3.xml
new file mode 100644
index 0000000..18d8663
--- /dev/null
+++ b/groups/multi-passenger/true/overlay-multi_passenger/audio/car_audio_configuration_vm3.xml
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (C) 2020 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+
+<!--
+  Defines the audio configuration in a car, including
+    - Audio zones
+    - Context to audio bus mappings
+    - Volume groups
+  in the car environment.
+-->
+<carAudioConfiguration version="2">
+    <zones>
+        <zone name="primary zone" isPrimary="true" occupantZoneId="0">
+            <volumeGroups>
+                <group>
+                    <device address="bus0_media_out">
+                        <context context="music"/>
+                        <context context="announcement"/>
+                        <context context="call"/>
+                    </device>
+                    <device address="bus6_notification_out">
+                        <context context="notification"/>
+                    </device>
+                </group>
+                <group>
+                    <device address="bus1_navigation_out">
+                        <context context="navigation"/>
+                    </device>
+                    <device address="bus2_voice_command_out">
+                        <context context="voice_command"/>
+                    </device>
+                </group>
+                <group>
+                    <device address="bus3_call_ring_out">
+                        <context context="call_ring"/>
+                    </device>
+                </group>
+                <group>
+                    <device address="bus5_alarm_out">
+                        <context context="alarm"/>
+                    </device>
+                    <device address="bus7_system_sound_out">
+                        <context context="system_sound"/>
+                        <context context="emergency"/>
+                        <context context="safety"/>
+                        <context context="vehicle_status"/>
+                    </device>
+                </group>
+	    </volumeGroups>
+            <inputDevices>
+		<inputDevice address="bus_1000_input_zone_0"/>
+            </inputDevices>
+        </zone>
+        <zone name="passenger seat zone" audioZoneId="1" occupantZoneId="1">
+            <volumeGroups>
+                <group>
+                    <device address="bus10_media_out">
+                        <context context="music"/>
+                        <context context="announcement"/>
+                        <context context="call"/>
+                    </device>
+                    <device address="bus16_notification_out">
+                        <context context="notification"/>
+                    </device>
+                </group>
+                <group>
+                    <device address="bus11_navigation_out">
+                        <context context="navigation"/>
+                    </device>
+                    <device address="bus12_voice_command_out">
+                        <context context="voice_command"/>
+                    </device>
+                </group>
+                <group>
+                    <device address="bus13_call_ring_out">
+                        <context context="call_ring"/>
+                    </device>
+                </group>
+                <group>
+                    <device address="bus15_alarm_out">
+                        <context context="alarm"/>
+                    </device>
+                    <device address="bus17_system_sound_out">
+                        <context context="system_sound"/>
+                        <context context="emergency"/>
+                        <context context="safety"/>
+                        <context context="vehicle_status"/>
+                    </device>
+                </group>
+            </volumeGroups>
+	    <inputDevices>
+		<inputDevice address="bus_2000_input_zone_1"/>
+            </inputDevices>
+        </zone>
+    </zones>
+</carAudioConfiguration>
+
diff --git a/groups/multi-passenger/true/product.mk b/groups/multi-passenger/true/product.mk
index 3ee8349..8a8f740 100644
--- a/groups/multi-passenger/true/product.mk
+++ b/groups/multi-passenger/true/product.mk
@@ -1,6 +1,20 @@
 # multi-passenger support
 PRODUCT_PACKAGE_OVERLAYS += $(LOCAL_PATH)/overlay-multi_passenger
 # Vendor audio configuration files
+ifeq ($(VM3), true)
+PRODUCT_COPY_FILES += \
+    $(LOCAL_PATH)/overlay-multi_passenger/audio/audio_policy_configuration_attached_devices.xml:vendor/etc/audio_policy_configuration_attached_devices.xml \
+    $(LOCAL_PATH)/overlay-multi_passenger/audio/audio_policy_configuration_devices.xml:vendor/etc/audio_policy_configuration_devices.xml \
+    $(LOCAL_PATH)/overlay-multi_passenger/audio/audio_policy_configuration_mixports.xml:vendor/etc/audio_policy_configuration_mixports.xml \
+    $(LOCAL_PATH)/overlay-multi_passenger/audio/audio_policy_configuration_routes.xml:vendor/etc/audio_policy_configuration_routes.xml \
+    $(LOCAL_PATH)/overlay-multi_passenger/audio/audio_policy_configuration_vm3.xml:vendor/etc/audio_policy_configuration.xml \
+    $(LOCAL_PATH)/overlay-multi_passenger/audio/car_audio_configuration_vm3.xml:vendor/etc/car_audio_configuration.xml
+else
 PRODUCT_COPY_FILES += \
     $(LOCAL_PATH)/overlay-multi_passenger/audio/audio_policy_configuration.xml:vendor/etc/audio_policy_configuration.xml \
     $(LOCAL_PATH)/overlay-multi_passenger/audio/car_audio_configuration.xml:vendor/etc/car_audio_configuration.xml
+endif
+
+ifeq ($(VM3), true)
+PRODUCT_PRODUCT_PROPERTIES += ro.vendor.audio.vm3=1
+endif
-- 
2.34.1

